openzwave-sys/openzwave-c/value_classes/value_id.cc

```diff
@@ -4,60 +4,69 @@
 extern "C" {
 
 // constructors
-ValueID * value_id_from_packed_id(uint32 home_id, uint64 id) {
-  return new ValueID(home_id, id);
+cValueID value_id_from_packed_id(uint32 home_id, uint64 id) {
+  cValueID vid(home_id, id);
+  return vid;
 }
-
-ValueID * value_id_from_values(uint32 home_id,
-                               uint8 node_id,
-                               ValueGenre genre,
-                               uint8 command_class_id,
-                               uint8 instance,
-                               uint8 value_index,
-                               ValueType type) {
-  return new ValueID(home_id, node_id, genre, command_class_id, instance, value_index, type);
+cValueID value_id_from_values(uint32 home_id,
+                            uint8 node_id,
+                            ValueGenre genre,
+                            uint8 command_class_id,
+                            uint8 instance,
+                            uint8 value_index,
+                            ValueType type) {
+  ValueID vid(home_id, node_id, genre, command_class_id, instance, value_index, type);
+  cValueID cvid(vid.GetHomeId(), vid.GetId());
+  return cvid;
 }
 
 // instance methods
-uint32 value_id_get_home_id(ValueID * value) {
-  return value->GetHomeId();
-}
-
-uint8 value_id_get_node_id(ValueID * value) {
-  return value->GetNodeId();
+uint8 value_id_get_node_id(cValueID *cvid) {
+  ValueID vid(cvid->home_id, cvid->id);
```

Comment:
and what we have now is more future-proof as it does not depend on anything private.